dashboard/src/app/index.module.ts

```diff
@@ -36,12 +36,81 @@ import IdeIFrameSvc from './ide/ide-iframe/ide-iframe.service';
 import {CheIdeFetcher} from '../components/ide-fetcher/che-ide-fetcher.service';
 import {RouteHistory} from '../components/routing/route-history.service';
 import {CheUIElementsInjectorService} from '../components/service/injector/che-ui-elements-injector.service';
-import {WorkspaceDetailsService} from './workspaces/workspace-details/workspace-details.service';
+import {OrganizationsConfig} from './organizations/organizations-config';
+import {TeamsConfig} from './teams/teams-config';
 
 // init module
 const initModule = angular.module('userDashboard', ['ngAnimate', 'ngCookies', 'ngTouch', 'ngSanitize', 'ngResource', 'ngRoute',
   'angular-websocket', 'ui.bootstrap', 'ui.codemirror', 'ngMaterial', 'ngMessages', 'angularMoment', 'angular.filter',
-  'ngDropdowns', 'ngLodash', 'angularCharts', 'uuid4', 'angularFileUpload']);
+  'ngDropdowns', 'ngLodash', 'angularCharts', 'uuid4', 'angularFileUpload', 'ui.gravatar']);
+
+declare const Keycloak: Function;
+function buildKeycloakConfig(keycloakSettings: any) {
+  return {
+    url: keycloakSettings['che.keycloak.auth_server_url'],
+    realm: keycloakSettings['che.keycloak.realm'],
+    clientId: keycloakSettings['che.keycloak.client_id']
+  };
+}
+interface IResolveFn<T> {
+  (value: T | PromiseLike<T>): Promise<T>;
+}
+interface IRejectFn<T> {
+  (reason: any): Promise<T>;
+}
+function keycloakLoad(keycloakSettings: any) {
+  return new Promise((resolve: IResolveFn<any>, reject: IRejectFn<any>) => {
+    const script = document.createElement('script');
+    script.async = true;
+    script.src = keycloakSettings['che.keycloak.auth_server_url'] + '/js/keycloak.js';
+    script.addEventListener('load', resolve);
+    script.addEventListener('error', () => reject('Error loading script.'));
+    script.addEventListener('abort', () => reject('Script loading aborted.'));
+    document.head.appendChild(script);
+  });
+}
+function keycloakInit(keycloakConfig: any) {
+  return new Promise((resolve: IResolveFn<any>, reject: IRejectFn<any>) => {
+    const keycloak = Keycloak(keycloakConfig);
+    keycloak.init({
+      onLoad: 'login-required', checkLoginIframe: false
+    }).success(() => {
+      resolve(keycloak);
+    }).error((error: any) => {
+      reject(error);
+    });
+  });
+}
+
+const keycloakAuth = {
+  isPresent: false,
+  keycloak: null,
+  config: null
+};
+initModule.constant('keycloakAuth', keycloakAuth);
+
+const promise = new Promise((resolve: IResolveFn<any>, reject: IRejectFn<any>) => {
+  angular.element.get('/api/keycloak/settings').then(resolve, reject);
+});
+promise.then((keycloakSettings: any) => {
+  keycloakAuth.config = buildKeycloakConfig(keycloakSettings);
+
+  // load Keycloak
+  return keycloakLoad(keycloakSettings).then(() => {
+    // init Keycloak
+    return keycloakInit(keycloakAuth.config);
+  }).then((keycloak: any) => {
+    keycloakAuth.isPresent = true;
```

Comment:
why do we need to store values in keycloakAuth object as it seems we don't keep a reference to that object ?

Reply:
@davidfestal @akurinnoy can you comment?

Reply:
@benoitf this object is referenced above:
```initModule.constant('keycloakAuth', keycloakAuth);```
and used by Keycloak interceptor, see [here](https://github.com/eclipse/che/blob/fb48c3d5d6942bd9fcc02f23f511c5b18743b35a/dashboard/src/components/interceptor/interceptor-config.ts) and [here](https://github.com/eclipse/che/blob/fb48c3d5d6942bd9fcc02f23f511c5b18743b35a/dashboard/src/components/interceptor/keycloak-token-interceptor.ts)

Reply:
yes sorry I failed with the text window of the review